src/Middleware/ImplicitHeadMiddlewareFactory.php

```diff
@@ -10,47 +10,45 @@
 namespace Zend\Expressive\Router\Middleware;
 
 use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamInterface;
 use Zend\Expressive\Router\Exception\MissingDependencyException;
 
-use const Zend\Expressive\Router\IMPLICIT_HEAD_MIDDLEWARE_RESPONSE;
-use const Zend\Expressive\Router\IMPLICIT_HEAD_MIDDLEWARE_STREAM_FACTORY;
-
 /**
  * Create and return an ImplicitHeadMiddleware instance.
  *
  * This factory depends on two other services:
  *
- * - IMPLICIT_HEAD_MIDDLEWARE_RESPONSE, which should resolve to a
- *   Psr\Http\Message\ResponseInterface instance.
- * - IMPLICIT_HEAD_MIDDLEWARE_STREAM_FACTORY, which should resolve to a
- *   callable that will produce an empty Psr\Http\Message\StreamInterface
- *   instance.
+ * - Psr\Http\Message\ResponseInterface, which should resolve to a callable
+ *   that will produce an empty Psr\Http\Message\ResponseInterface instance.
+ * - Psr\Http\Message\StreamInterface, which should resolve to a callable
+ *   that will produce an empty Psr\Http\Message\StreamInterface instance.
```

Comment:
I'm on the fence about using the message interfaces as the service names, particularly as they will resolve to callables instead of instances of those interfaces.

That said, if we use anything else, we have to make up names, and likely use constants to refer to them, which leads to dependency issues. 

As such, the interface names are likely best, until PSR-17 is ratified. That can wait until the next major version; users will still be able to use PSR-17 internally, by resolving the PSR-7 interface name services to specific PSR-17 instance methods or closures around the factories.